lang/cs/Org.Apache.REEF.Tests/Functional/IMRU/IMRUCloseTaskTest.cs

```diff
@@ -30,24 +34,22 @@ namespace Org.Apache.REEF.Tests.Functional.IMRU
 {
     /// <summary>
     /// This is to test close event handler in IMRU tasks
-    /// The test provide IRunningTask, IFailedTask and ICOmpletedTask handlers so that to trigger close events and handle the 
+    /// The test provide IRunningTask, IFailedTask and ICompletedTask handlers so that to trigger close events and handle the 
     /// failed tasks and completed tasks
     /// </summary>
     [Collection("FunctionalTests")]
     public class IMRUCloseTaskTest : IMRUBrodcastReduceTestBase
     {
         private const string CompletedTaskMessage = "CompletedTaskMessage";
+        private const string FailEvaluatorMessage = "FailEvaluatorMessage";
         private const string FailTaskMessage = "FailTaskMessage";
 
         /// <summary>
         /// This test is for running in local runtime
         /// It sends close event for all the running tasks.
-        /// It first informs the Call method to stop.
-        /// If Call method is running properly, it will respect to this flag and will return properly, that will end up ICompletedTask event.
-        ////If Call method is hung some where and cannot be returned, the close handler will throw exception, that would cause IFailedTask event.
-        /// As we are testing IMRU Task not a test task, the behavior is not deterministic. It can be CompletedTask or FailedTask
-        /// No matter how the task is closed, the total number of completed task and failed task should be equal to the 
-        /// total number of the tasks.
+        /// In the task close handler, the cancellation token will be set, resulting the task
+        /// returned from the Call() method. 
+        /// Expect all the 4 tasks are completed. No Failed task and no failed Evaluator. 
```

Comment:
The exception might happen..Consider the case where one node receives close message and closes its connection...The other node might get the exceptions because of that and hence task will throw exception. I understand that currently while reading from stream we are not propagating error messages to operator callers, so they will remain blocked and hence cancellation token will help then come out. But what if they try to write...Then there will definitely be exception. Any case we are also trying to address this blocking issue. So sooner or later you are going to hit that.